src/core/game/Game.ts

```diff
@@ -302,6 +302,7 @@ export class PlayerInfo {
   public readonly clan: string | null;
 
   constructor(
+    public readonly pattern: string | null,
```

Comment:
Please use `undefined` instead of `null` to keep it consistent with flag.

```suggestion
    public readonly pattern: string | undefined,
```